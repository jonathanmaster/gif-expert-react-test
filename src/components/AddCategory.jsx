import React from "react";
import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    //La expresión inputValue.trim().length se refiere a la longitud del valor de la variable "inputValue" después de eliminar los espacios en blanco iniciales y finales utilizando el método trim().
    if (inputValue.trim().length <= 1) return;

    // setCategories(categories => [inputValue, ...categories])
    setInputValue("");
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form" className="container-form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
