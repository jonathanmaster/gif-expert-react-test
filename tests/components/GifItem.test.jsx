import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en el componente <GifItem/>', () => {

    const title = 'Saitam'
    const url = 'https://one-punch.com/saitama.jpg'

    test('debe de hacer match con el snapshot', () => {

        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el url y el alt indicado', () => {
        render(<GifItem title={title} url={url}/>)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)

    });

    // es sobre la etiqueta 'p'
    // test('debe de mostrar el titul en el componente', () => {

    //     render(<GifItem title={title} url={url}/>)
    //     expect(screen.getAllByText(title)).toBeTruthy()
    // })


})




