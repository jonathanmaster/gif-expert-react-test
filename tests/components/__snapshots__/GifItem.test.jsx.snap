// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitam"
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
